### DISCUSSION TITLE
initial status set to "loading"problem

### DISCUSSION THREAD

**Discussion by MoeHamdan** [TanStack/query#6112] [Q&A] [answered]
### Describe the bug

The initial status value is "loading", this is a problem when initially the query is disabled and then enabled later on. Or when in some cases it will be enabled initially or disabled. 

I have a loading backdrop that depends on the status of the query, this will make it always shown, even if the query did not execute.

### Your minimal, reproducible example

N/a

### Steps to reproduce

  var providerResult = useQuery(
    ["getProviderById", id],
    () => getProviderById(parseInt(id!)),
    {
      refetchOnWindowFocus: false,
      enabled: existing,
    }
  );

existing might be false/true depending on different scenarios.

### Expected behavior

I would expect to either have the old status "Idle" or add new status call it "initial" or the like. It does not make sense to have the intial status "loading" while the query did not execute at all.

### How often does this bug happen?

None

### Platform

any browser

### Tanstack Query adapter

None

### TanStack Query version

4.35.7

**Comment by TkDodo** [comment, marked as answer]
it is documented that it starts in `status: 'loading'` here:

https://tanstack.com/query/v4/docs/react/guides/dependent-queries#usequery-dependent-query

> I would expect to either have the old status "Idle" or add new status call it "initial" or the like. It does not make sense to have the intial status "loading" while the query did not execute at all.

There won't be a 4th state because that complicates states for every case where you're not using `enabled`. The rationale is documented here: https://tanstack.com/query/v4/docs/react/guides/queries#why-two-different-states

In v5, the `loading` state is renamed to `pending` to make it less confusing. `pending` can hopefully be better associated with the fact that we don't have any data, which is true for queries that are not enabled, too.

The flag you want to check for is `isInitialLoading` (which, in v5, will be `isLoading`)

**Comment by MoeHamdan** [comment]
Thanks @TkDodo for your comprehensive feedback. I think fetchStatus will solve my problem. All I need is to know if the api is being called .
